[IV-23-24] Objetivo 2

danibarranqueroo opened the pull request:
## Objetivo2
### Sobre la estructura del repositorio

* [x] ¿He seguido las mejores prácticas en el nombre de las clases y ficheros y
   disposición de los mismos?
* [x] ¿He mirado tanto reglas generales (para cualquier repo) como reglas
específicas para el lenguaje con el que se está trabajando?

### Sobre el planteamiento

* [x] ¿Se han planteado una serie de issues, asignados a las historias de
usuario pertinentes, que clarifiquen qué es lo que necesitan los usuarios e
identifiquen las diferentes partes del problema?
* [x] ¿Los issues representan un problema, y no una tarea?

### Sobre el análisis del problema

* [x] ¿Se ha documentado qué análisis se ha hecho sobre el dominio para decir lo
   que se ha creado?
* [x] ¿Se ha documentado por qué se ha elegido que lo creado sea un objeto valor,
   una entidad o un agregado?
* [x] ¿He propuesto un producto mínimamente viable, que en muchos casos será un solo
   objeto valor que no dependa de ningún otro (y que sea la base de muchos
   otros)?.

### Sobre el código

* [x] ¿Todos los mensajes de commit explican el cambio, y no se
      limitan a repetir el nombre del fichero que se ha cambiado?
* [x] ¿Los mensajes de commit siguen el formato estándar y buenas prácticas?
* [x] ¿Se ha hecho una revisión real del código para comprobar que todos
   los atributos y funciones creadas están respaldadas por una HU?
* [x] ¿Todos los issues creados están relacionados con a una HU?
* [x] ¿Ha asignado el propietario/la propietaria del repositorio todos los
issues planteados al *milestone* correspondiente?
* [x] ¿Todos los cambios en el código están asignados a un issue al que se
      referencia en un commit?
* [x] ¿Se ha asignado al mismo milestone el PR que se ha hecho?
* [x] ¿Es el milestone sobre el que estoy trabajando un producto
   mínimamente viable? ¿O tengo que solicitar al product manager que precise de
   qué producto se trata?
* [x] ¿Se ha comprobado que el código entregado sea sintácticamente
correcto?

danibarranqueroo committed:
Creación de yaml y cambio en el gitignore

danibarranqueroo committed:
Cambio en el yaml

danibarranqueroo committed:
Debatido lenguaje en #4, closes #4

danibarranqueroo commented:
Hola @raultl12 en este pr es donde iremos trabajando, cuando puedas enlázalo a tu milestone 0 y añade el tag de hacktoberfest-accepted.
Cualquier duda o problema que tengas sobre lo que voy haciendo puedes comentar sin problema!

danibarranqueroo committed:
Creación de dos clases para la division del problema, closes #7

danibarranqueroo committed:
Estructura de la clase usuario y sus propiedades, closes #8

danibarranqueroo committed:
Estructura de la clase habilidad y sus propiedades, closes #9

danibarranqueroo committed:
Guardamos los tipos de rol como un enum, closes #10

danibarranqueroo committed:
Guardamos los tipos de actitud como un enum, closes #11

danibarranqueroo committed:
Guardamos el ratio como number ya que proporciona la precision necesario, closes #12

danibarranqueroo committed:
Guardamos los tipos de rango como un enum y finalizamos clase usuario, closes #13

danibarranqueroo committed:
Guardamos los tipos de habilidad como un enum, closes #14

danibarranqueroo committed:
Guardamos los tipos de nivel como un enum, closes #15

danibarranqueroo committed:
Cambio el fichero iv.yaml

danibarranqueroo commented:
Buenas tardes @raultl12! Cuando puedas revisa un poco todo lo que he ido haciendo y si algo no lo ves correcto o no te gusta la forma en la que lo he enfocado comenta sin problema tanto en los issues, como aquí o en los ficheros y vamos así perfeccionando el objetivo. Cuando lo veas listo acuérdate de poner por aquí la lista de comprobación :raised_hands:.

raultl12 commented on src/Habilidad.ts:
```diff
@@ -0,0 +1,25 @@
+// Clase Habilidad
+
+export enum TipoHabilidad {
+    Posicionamiento = "Posicionamiento",
```
Hola @danibarranqueroo. El codigo lo veo bien, aunque la forma de escribir los enums creo que deberia ser distinta. En lugar de poner Posicionamiento = "Posicionamiento", deberia de aparecer solo Posicionamiento. TS ya se encarga internamente de asignar enteros de forma incremental a los valores de los enumerados (0, 1, 2...).

danibarranqueroo replied:
Perfecto entonces, acabo de arreglarlo en el commit [5799f8c](https://github.com/raultl12/TeamFinder/pull/5/commits/5799f8cadb8dcbe8fecfd418e7dbcc667797f038)

raultl12 commented on src/Habilidad.ts:
```diff
@@ -0,0 +1,25 @@
+    public nivel: Nivel;
+  
+    constructor(tipo: TipoHabilidad, nivel: Nivel) {
+      this.tipo = tipo;
```
En cuanto a los constructores, mirando los [errores frecuentes](https://github.com/JJ/IV-/blob/master/errores/errores-objetivo-2.md), creo que habria que eliminarlos.
>Si el constructor no añade nada a la clase o struct definido, es mejor no hacerlo; por ejemplo, si simplemente inicializa las variables con respecto a lo que se le pasa como argumento.

danibarranqueroo replied:
Gracias por estar atento a ello, no me había dado cuenta, resuelto ya en el commit [28e4c2b](https://github.com/raultl12/TeamFinder/pull/5/commits/28e4c2bf2ad43bcf6305ec51c4995d39f0bfa9a8)

danibarranqueroo committed:
Arreglo todos los enum

danibarranqueroo committed:
Elimino los constructores

danibarranqueroo commented:
Buenos días @raultl12 ya he reaizado los cambios propuestos, si no hay ningún otro problema o cosa que piensas que podríamos hacer de otra forma confírmame y procedo a poner este pr para revisión.

raultl12 reviewed (commented):
Hola @danibarranqueroo. Lo veo bien para pedir revisión

raultl12 commented:
* [x] ¿Los issues que ha planteado la persona son efectivamente problemas y
evidencian comprensión de las historias de usuario? Si no es así, es mejor que
se le transmita antes de que empiece a crear cualquier código erróneo.
* [x] ¿Has comprobado que efectivamente el compañero/a ha seguido *su* lista de
      comprobación?
* [x] ¿El código creado es una modelización válida de las historias de usuario
      creadas por ti, y te sirve como base para programar la lógica de negocio?
* [x] ¿El código creado sigue las mejores prácticas para el lenguaje de
      programación que hayáis consensuado, tanto en identificadores, como
      nombres de ficheros, como disposición de los diferentes ficheros en el
      repositorio? Por ejemplo, ¿todos los commits se refieren a un issue y
      aportan código para resolver el problema planteado en el mismo?
* [x] ¿He comprobado que el código compile, o que quien lo ha programado ha lo
ha hecho y documentado en el repositorio?
* [x] ¿He añadido el PR y todos los issues que se resuelvan en él al milestone
correspondiente, que será el primero?
* [x] ¿Cubre totalmente el milestone 0, o hará falta modelización adicional en
el objetivo 4 para poder programar la lógica de negocio?
* [x] En resumen, ¿considero que con lo hecho puedo ponerme a trabajar
directamente en el siguiente milestone, que incluye la lógica de negocio?

carlosservi commented:
Buenas, me ha tocado ser revisor. He estado mirando todo el trabajo realizado y lo veo bastante bien.

danibarranqueroo commented:
Buenas @JJ ya han pasado 2 días desde que lo aprobó el compañero y ningún otro revisor se ha puesto en contacto, por si puede revisarlo.

shvtwp commented:
Hola @danibarranqueroo , he estado revisando y creo que el nombre de la carpeta src debería ser lib. Además, de cara a un futuro te recomendaría echarle un vistazo a las buenas prácticas para escribir un commit. Por lo demás, no veo nada más que mejorar.

JJ commented on iv.yaml:
```diff
@@ -0,0 +1,2 @@
+lenguaje: TypeScript
+entidad: src/Usuario.ts
```
No se usan mayúsculas en los repositorios. Todos los nombres de fichero deben ir en minúsculas.

JJ reviewed (changes requested):
Aparte de lo comentado en los issues específicos, piensa que es imposible programar la lógica de negocio si no tienes ninguna entidad que la maneje y pueda gestionar los objetos valor. Te agradecería que lo hicieras con cierta rapidez, y también contemples lo indicado por @shvtwp, porque en general es una práctica más moderna que usar src.

danibarranqueroo committed:
Nombres en minúscula y sustituyo src por lib

danibarranqueroo committed:
Creación de entidad y actualizao iv.yaml

Es necesario una entidad que gestione Usuario y Habilidad así que creo Entrenamiento para ello

danibarranqueroo commented:
Gracias por la ayuda, diría que ya está listo de nuevo para revisión @JJ.

JJ commented:
@danibarranqueroo no puedes crear una entidad de la nada, sin un issue que la guíe, como has hecho en [5468c21](https://github.com/raultl12/TeamFinder/pull/5/commits/5468c2190180a73e4bc1436a62ae0d2ae10c68e4) Voy a revisarlo de todas formas por la premura que corre, pero sin seguir una metodología precisa que pase de HU a través de issue al código (y documentándolo en el mensaje de commit) es prácticamente imposible hacerlo bien, y simplemente imposible evaluar si el código es correcto.

JJ commented on lib/entrenamiento.ts:
```diff
@@ -0,0 +1,10 @@
+// Clase Entrenamiento
+
+import { Usuario } from './usuario'; // Importar la clase Usuario
```
¿Crees que los comentarios aportan algo?

danibarranqueroo replied:
Solucionado en [271a07d](https://github.com/raultl12/TeamFinder/pull/5/commits/271a07d8e5fccfadf86afedb05292c235d8da2a9)

JJ commented on lib/habilidad.ts:
```diff
@@ -0,0 +1,20 @@
+    Apoyo,
+  }
+
+export enum Nivel {
```
Con un commit como 5799f8c es imposible saber si esto está bien o mal. Si los "arreglas" es porque tienen algún problema, si tienen algún problema debería estar en un issue, y ese issue debería referirse a una historia de usuario.

danibarranqueroo replied:
Solucionado en [fbfd265](https://github.com/raultl12/TeamFinder/pull/5/commits/fbfd2658623ce87b9addac8751f8e8d4abbc2e69)

JJ reviewed (commented):
Creo que hay un problema de base, que es que la HU no deja demasiado claro de qué va a ir la lógica de negocio, con lo que es muy complicado. La clave de la HU #2 es la palabra "entrenamiento" y es lo que hay que modelar, pero para una persona ajena a este mundo es muy complicado entender en qué consiste un entrenamiento.
Creo que @danibarranqueroo , a pesar de eso, ha intentado hacer un buen trabajo modelando al menos los tipos de jugadores y si hubiera seguido una metodología adecuada (con issues que describan problemas, por ejemplo, y con mensajes de commit que expliquen en qué issue se está trabajando) mejoraría bastante, pero lo esencial de este objetivo (y de todos los demás) es aplicar correctamente una metodología, así que conviene que se creen nuevos issues o se reescriban para poder seguirla correctamente.

JJ commented:
Vamos a dejar la clase entrenamiento para que la haga @raultl12 ; por favor corrige el resto. Y con un poco de prisa, por favor.

danibarranqueroo committed:
Elimino clase entrenamiento, closes #43

La clase entrenamiento se modelará en el objetivo 4 por lo que ahora no es necesario que la cree

danibarranqueroo committed:
Elimino comentarios, closes #44

Los comentarios no aportaban nada por lo que mejor se eliminan

danibarranqueroo committed:
Cambio fichero iv.yaml, closes #45

Al eliminar el fichero entrenamiento este fichero estaba mal por lo que había que actualizarlo

danibarranqueroo committed:
Creo fichero para los enums, closes #48

Los enums que estaban en usuario y habilidad se pasan a un fichero aparte

danibarranqueroo commented:
Listo para revisión @JJ.

JJ reviewed (approved):
👍

JJ commented:
Por favor, empieza inmediatamente con el objetivo 3 y entrégalo en los próximos días. Si puedes antes del lunes.
